src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/CircleSelection.js

```diff
@@ -1,88 +1,233 @@
+                radius,
+            })
+        );
+    }
+
+    centerSelection(selection: SelectionData): SelectionData {
+        const {containerWidth, containerHeight} = this.props;
+
+        const halfWidth = containerWidth / 2;
+        const halfHeight = containerHeight / 2;
+
+        return {
+            ...selection,
+            left: halfWidth,
+            top: halfHeight,
+        };
+    }
+
+    handleCircleDoubleClick = () => {
+        const {onChange, resizable} = this.props;
+
+        if (resizable) {
+            onChange(this.maximumSelection);
+
+            return;
+        }
+
+        onChange(this.normalize(this.centerSelection(this.value)));
+    };
+
+    handleCircleChange = (value: SelectionData) => {
+        const {onChange} = this.props;
+
+        onChange(this.normalize(value));
+    };
+
     render() {
-        const {
-            children,
-            containerHeight,
-            containerWidth,
-            disabled,
-            label,
-            maxRadius,
-            minRadius,
-            onChange,
-            resizable,
-            round,
-            skin,
-            usePercentageValues,
-            value,
-        } = this.props;
+        const {disabled, label, onFinish, resizable, skin} = this.props;
+        const {left, top, radius} = this.value;
+
+        return (
+            <ModifiableCircle
+                disabled={disabled}
+                label={label}
+                left={left}
+                onChange={this.handleCircleChange}
+                onDoubleClick={this.handleCircleDoubleClick}
+                onFinish={onFinish}
+                radius={radius}
+                resizable={resizable}
+                skin={skin}
+                top={top}
+            />
+        );
+    }
+}
+
+const CircleSelectionWrapper = withPercentageValues(CircleSelectionComponent);
+
+class CircleSelectionContainer extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        maxRadius: undefined,
+        minRadius: undefined,
+        resizable: true,
+        round: true,
+        skin: 'outlined',
+        usePercentageValues: false,
+    };
+
+    render() {
+        const {children, ...rest} = this.props;
 
         return (
             <div className={circleSelectionStyles.selection}>
                 {children}
-                <CircleSelectionRenderer
-                    containerHeight={containerHeight}
-                    containerWidth={containerWidth}
-                    disabled={disabled}
-                    label={label}
-                    maxRadius={maxRadius}
-                    minRadius={minRadius}
-                    onChange={onChange}
-                    resizable={resizable}
-                    round={round}
-                    skin={skin}
-                    usePercentageValues={usePercentageValues}
-                    value={value}
-                />
+                <CircleSelectionWrapper {...rest} />
```

Comment:
But then there are four possibilities:
* CircleSelection
* CircleSelectionWithContainerSize
* CircleSelectionWithPercentageValues
* CircleSelectionWithPercentageValuesWithContainerSize

That's kind of confusing either, isn't it?

Reply:
But there is another problem: The export for the tests need to be either with percentage values or without, it also needs to be either rendering children if there are any or not, if there aren't, but NOT wrapping with withContainerSize, even if there are children. So that makes that whole struct a lot more complicated 😄

Reply:
Also, the component which is rendering the `CircleSelectionComponent` along with the `children` should still be named `CircleSelectionContainer` IMO, because that's what it is, a container containing the `CircleSelectionComponent` and the `children`

Reply:
If I am not wrong the resulting component with my suggestion would accept a `usePercentageValues` prop, so the tests would not have to care about the `withPercentageValues` HOC at all, because from an external perspective it would work exactly as it did before.

Regarding the `withContainerSize`, that sounds even more like a reason to not export the component without it, and mock the `withContainerSize` HOC properly in the tests instead.

Regarding the last naming, then I would prefer `CircleSelectionWrapper`, because the term `Wrapper` does not have this ambiguity.